src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,265 @@
+import styled from "styled-components"
+import { left, LeftProps, right, RightProps } from "styled-system"
+import { Media } from "Utils/Responsive"
+
+export interface FlickitySettings {
+  draggable?: boolean
+  freeScroll?: boolean
+  wrapAround?: boolean
+  groupCells?: boolean | number | string
+  autoPlay?: boolean | number
+  fullscreen?: boolean
+  fade?: boolean
+  adaptiveHeight?: boolean
+  watchCSS?: boolean
+  hash?: boolean
+  dragThreshold?: number
+  friction?: number
+  selectedAttraction?: number
+  freeScrollFriction?: number
+  imagesLoaded?: boolean
+  lazyLoad?: boolean | number
+  bgLazyLoad?: boolean | number
+  initialIndex?: number
+  accessibility?: boolean
+  setGallerySize?: boolean
+  resize?: boolean
+  cellAlign?: "left" | "center" | "right"
+  contain?: boolean
+  percentPosition?: boolean
+  rightToLeft?: boolean
+  prevNextButtons?: boolean
+  pageDots?: boolean
+}
+
+type Arrow = (
+  props: {
+    Arrow: React.ReactType
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  settings?: FlickitySettings
+}
+
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+
+  render() {
+    return (
+      <>
+        <Media at="xs">
+          <SmallCarousel {...this.props} />
+        </Media>
+        <Media greaterThan="xs">
+          <LargeCarousel {...this.props} />
+        </Media>
+      </>
+    )
+  }
+}
+
+/*
+ * Returns null if no window for SSR
+ */
+const Flickity =
+  typeof window !== "undefined"
+    ? require("react-flickity-component")
+    : () => null
+
+export const LargeCarousel = (props: Props) => {
+  const [isMounted, toggleMounted] = useState(false)
+  const [currentSlideIndex, setSlideIndex] = useState(0)
+  const [lastItemVisible, setLastItemVisible] = useState(false)
+
+  useEffect(() => {
+    toggleMounted(true)
+  }, [])
+  const flicktyRef = useRef(null)
+
+  let flickity = null
+
+  const checkLastItemVisible = () => {
+    if (!flickity.slides.length) {
+      return
+    }
+    const isLastItemVisible = flickity.selectedElements.includes(
+      flickity.getLastCell().element
+    )
+    setLastItemVisible(isLastItemVisible)
+  }
+
+  if (isMounted) {
+    if (flicktyRef.current !== null) {
+      flickity = flicktyRef.current.flkty
+    }
+
+    flickity.on("select", index => {
+      setSlideIndex(index)
+      checkLastItemVisible()
+    })
+  }
+
+  const settings = {
+    draggable: false,
+    freeScroll: false,
+    wrapAround: false,
+    cellAlign: "left",
+    pageDots: false,
+    prevNextButtons: false,
+    groupCells: true,
+    contain: true,
+    ...props.settings,
+  }
+
+  const LeftArrow = () => {
+    const getFlickity = flickity
```

Comment:
Why add an intermediary variable? Seems to introduce some unnecessary indirection.